_docs/concepts/network-and-auth/auth.md

```diff
@@ -0,0 +1,150 @@
+
+1.  [Istio Manager](https://istio.io/docs/concepts/traffic-management/manager.html) generates the config with proper cert/key and secure naming information, which defines what service account(s) can run a certain service, and passes it to Envoy. 
+
+### Runtime Phase
+
+
+
+1.  The outbound traffic from a client service is rerouted to its local Envoy. 
+
+1.  The client side Envoy starts a mutual TLS handshake with the server side Envoy. During the handshake, it also does a secure naming check to verify that the service account presented in the server certificate can run the server service. 
+
+1.  The traffic is forwarded to the server side Envoy after mTLS connection is established, which is then forwarded to the server service through local TCP connections.
+
+## Service to Service Auth Best Practice
+
+In this section, we provide a few deployment guidelines and then discuss a real-world scenario. 
+
+### Deployment Guidelines
+
+
+
+*   If there are multiple service operators (a.k.a. [SREs](https://en.wikipedia.org/wiki/Site_reliability_engineering)) deploying different services in a cluster (typically in a medium- or large-size cluster), we recommend creating a separate [namespace](https://kubernetes.io/docs/tasks/administer-cluster/namespaces-walkthrough/) for each SRE team to isolate their access. For example, you could create a "team1-ns" namespace for team1, and "team2-ns" namespace for team2, such that both teams won't be able to access each other's services.
+
+*   If Istio CA is compromised, all its managed keys & certs in the cluster may be exposed. We *strongly* recommend running Istio CA on a dedicated namespace (for example, istio-ca-ns), which only cluster admins have access to.
+
+### Example
+
+Let's consider a 3-tier application with three services: photo-frontend, photo-backend, and datastore. Photo-frontend and photo-backend services are managed by the photo SRE team while the datastore service is managed by the datastore SRE team. Photo-frontend can access photo-backend, and photo-backend can access datastore. However, photo-frontend cannot access datastore.
+
+In this scenario, a cluster admin can creates 3 namespaces: istio-ca-ns, photo-ns, and datastore-ns. Admin has access to all namespaces, and each team only has access to its own namespace. The photo SRE team creates 2 service accounts to run photo-frontend and photo-backend respectively in namespace photo-ns. The datastore SRE team creates 1 service account to run the datastore service in namespace datastore-ns. Moreover, we need to enforce the service access control in [Istio Mixer](https://istio.io/docs/concepts/policy-and-control/mixer.html) such that photo-frontend cannot access datastore.
+
+In this setup, Istio CA is able to provide key/cert management for all namespaces, and successfully prevent any team from messing up services run by other teams.
+
+## Future Work
```

Comment:
You've already mentioned future features in the intro above. I'd drop this whole section.